core/keyserver/revisions_test.go

```diff
@@ -119,30 +113,21 @@ func MustEncodeToken(t *testing.T, low time.Time) string {
 
 func TestListMutations(t *testing.T) {
 	ctx := context.Background()
-	fakeBatches := batchStorage{
-		1: SourceList{{LogId: 0, LowestInclusive: 2, HighestExclusive: 7}},
-		2: SourceList{{LogId: 0, LowestInclusive: 7, HighestExclusive: 11}},
+	dirID := "TestListMutations"
+	logID := int64(0)
+	fakeLogs := memory.NewMutationLogs()
+	idx := make([]time.Time, 0, 12)
+	for i := int64(0); i < 12; i++ {
+		ts, err := fakeLogs.Send(ctx, dirID, logID, genEntryUpdates(t, i, i+1)...)
+		if err != nil {
+			t.Fatal(err)
+		}
+		idx = append(idx, ts)
 	}
 
-	fakeLogs := make(mutations)
-	for _, sources := range fakeBatches {
-		for _, source := range sources {
-			extendBy := source.HighestExclusive - int64(len(fakeLogs[source.LogId]))
-			if extendBy > 0 {
-				fakeLogs[source.LogId] = append(fakeLogs[source.LogId], make([]*mutator.LogMessage, extendBy)...)
-			}
-			for i := source.LowestInclusive; i < source.HighestExclusive; i++ {
-				fakeLogs[source.LogId][i] = &mutator.LogMessage{
-					ID: time.Unix(0, i*int64(time.Nanosecond)),
-					Mutation: &pb.SignedEntry{
-						Entry: mustMarshal(t, &pb.Entry{
-							Index:      []byte(fmt.Sprintf("key_%v", i)),
-							Commitment: []byte(fmt.Sprintf("value_%v", i)),
-						}),
-					},
-				}
-			}
-		}
+	fakeBatches := batchStorage{
+		1: SourceList{{LogId: 0, LowestInclusive: idx[2].UnixNano(), HighestExclusive: idx[7].UnixNano()}},
+		2: SourceList{{LogId: 0, LowestInclusive: idx[7].UnixNano(), HighestExclusive: idx[11].UnixNano()}},
 	}
 
 	for _, tc := range []struct {
```

Comment:
`token` needs to be a string because that's the input that the API takes, and we have a line for invalid input.
`wantNext` would be silly to be a string because I'd just unmarshal it in the test.